terraform/deployments/monitoring-test/grafana/main.tf

```diff
@@ -0,0 +1,135 @@
+terraform {
+  backend "s3" {
+    bucket  = "govuk-terraform-test"
+    key     = "projects/grafana.tfstate"
+    region  = "eu-west-1"
+    encrypt = true
+  }
+}
+
+provider "aws" {
+  region = "eu-west-1"
+
+  assume_role {
+    role_arn = var.assume_role_arn
+  }
+}
+
+provider "random" {}
+
+data "aws_region" "current" {}
+
+# TODO pass this ARN in from the govuk deployment (via terraform remote state) instead of using a data source
+data "aws_iam_role" "execution" {
+  name = "fargate_execution_role"
+}
+
+# TODO pass this ARN in from the govuk deployment (via terraform remote state) instead of using a data source
+data "aws_iam_role" "task" {
+  name = "grafana_task_role"
+}
+
+data "terraform_remote_state" "govuk" {
+  backend   = "s3"
+  workspace = terraform.workspace
+  config = {
+    bucket   = "govuk-terraform-test"
+    key      = "projects/govuk.tfstate"
+    region   = data.aws_region.current.name
+    role_arn = var.assume_role_arn
+  }
+}
+
+data "terraform_remote_state" "monitoring" {
+  backend = "s3"
+  config = {
+    bucket   = "govuk-terraform-test"
+    key      = "projects/monitoring.tfstate"
+    region   = "eu-west-1"
+    role_arn = var.assume_role_arn
+  }
+}
+
+data "aws_secretsmanager_secret" "github_client_id" {
+  name = "grafana_github_client_id"
+}
+
+data "aws_secretsmanager_secret" "github_client_secret" {
+  name = "grafana_github_client_secret"
+}
+
+
+resource "random_string" "grafana_password" {
```

Comment:
Cool! I think we should use `random_password` here

https://registry.terraform.io/providers/hashicorp/random/latest/docs/resources/password

Reply:
good catch, I was going for that but I don't know what happened.